Packs/MandiantAdvantageThreatIntelligence/Integrations/ThreatIntelligence/ThreatIntelligence.py

```diff
@@ -0,0 +1,1426 @@
+) -> List:
+    """
+    Get a list of new indicators
+    Args:
+        client (MandiantClient): client
+        last_run (str): last run as free text or date format
+        indicator_type (str): the desired type to fetch
+        limit (int): number of indicator to fetch
+    Returns:
+        List: A list of new indicators
+    """
+    start_date = arg_to_datetime(last_run)
+    minimum_mscore = int(demisto.params().get("feedMinimumConfidence", 80))
+    exclude_osint = demisto.params().get("feedExcludeOSIntel", True)
+
+    params = {}
+    if indicator_type == "Indicators":
+        # for indicator type the earliest time to fetch is 90 days ago
+        earliest_fetch = arg_to_datetime("90 days ago")
+        assert earliest_fetch is not None
+
+        param_start_date: datetime = datetime.fromtimestamp(0)
+        if start_date is not None:
+            param_start_date = max(
+                earliest_fetch.replace(tzinfo=pytz.UTC), start_date.replace(tzinfo=pytz.UTC)
+            )  # type:ignore
+        else:
+            param_start_date = earliest_fetch
+        params = {
+            "start_epoch": int(param_start_date.timestamp()),
+            "limit": limit,
+            "gte_mscore": minimum_mscore,
+            "exclude_osint": exclude_osint,
+        }  # type:ignore
+
+    new_indicators_list = client.get_indicators(indicator_type, params=params)
+
+    if indicator_type != "Indicators":  # new to old
+        new_indicators_list.sort(key=get_last_updated, reverse=True)  # type:ignore
+        new_indicators_list = list(
+            filter(last_updated_filter(start_date), new_indicators_list)  # type: ignore
+        )
+
+    return new_indicators_list
+
+
+def get_indicator_list(
+    client: MandiantClient, limit: int, first_fetch: str, indicator_type: str
+) -> List[Dict]:
+    """
+    Get a list of indicators of the given type
+    Args:
+        client (MandiantClient): client
+        limit (int): number of indicators to return.
+        first_fetch (str): Get indicators newer than first_fetch.
+        indicator_type (str): indicator type
+    Returns:
+        List[Dict]: list of indicators
+    """
+    last_run_dict = demisto.getLastRun()
+    indicators_list = last_run_dict.get(f"{indicator_type}List", [])
+
+    if len(indicators_list) < limit:
+        last_run = last_run_dict.get(indicator_type + "LastFetch", first_fetch)
+        new_indicators_list = get_new_indicators(
+            client, last_run, indicator_type, limit
+        )
+        indicators_list += new_indicators_list
+
+    if indicators_list:
+        new_indicators_list = indicators_list[:limit]
+        last_run_dict[indicator_type + "List"] = indicators_list[limit:]
+        date_key = "last_seen" if indicator_type == "Indicators" else "last_updated"
+        last_run_dict[indicator_type + "LastFetch"] = new_indicators_list[-1][date_key]
+
+        demisto.setLastRun(last_run_dict)
```

Comment:
setting the LastRun object with a updated time should happen only in the fetch-indicators command and currently happens also in the `mati-feed-get-indicators` command which is not good and will cause missing indicators in the fetch process